dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ExecutorServiceImpl.java

```diff
@@ -686,89 +705,117 @@ private int createCommand(CommandType commandType, long processDefineCode,
-                command.setCommandParam(JSONUtils.toJsonString(cmdParam));
-                createCount = processService.createCommand(command);
-
-                // dependent process definition
-                List<Schedule> schedules = processService.queryReleaseSchedulerListByProcessDefinitionCode(command.getProcessDefinitionCode());
-
-                if (schedules.isEmpty() || complementDependentMode == ComplementDependentMode.OFF_MODE) {
-                    logger.info("process code: {} complement dependent in off mode or schedule's size is 0, skip "
-                            + "dependent complement data", command.getProcessDefinitionCode());
-                } else {
-                    dependentProcessDefinitionCreateCount += createComplementDependentCommand(schedules, command);
+                if(startDate != null && endDate != null){
+                    cmdParam.put(CMDPARAM_COMPLEMENT_DATA_START_DATE, startDate);
+                    cmdParam.put(CMDPARAM_COMPLEMENT_DATA_END_DATE, endDate);
+                    command.setCommandParam(JSONUtils.toJsonString(cmdParam));
+                    createCount = processService.createCommand(command);
+
+                    // dependent process definition
+                    List<Schedule> schedules = processService.queryReleaseSchedulerListByProcessDefinitionCode(command.getProcessDefinitionCode());
+
+                    if (schedules.isEmpty() || complementDependentMode == ComplementDependentMode.OFF_MODE) {
+                        logger.info("process code: {} complement dependent in off mode or schedule's size is 0, skip "
+                                + "dependent complement data", command.getProcessDefinitionCode());
+                    } else {
+                        dependentProcessDefinitionCreateCount += createComplementDependentCommand(schedules, command);
+                    }
                 }
-
                 break;
             }
             case RUN_MODE_PARALLEL: {
-                if (start.after(end)) {
-                    logger.warn("The startDate {} is later than the endDate {}", start, end);
-                    break;
-                }
-
-                List<Date> listDate = new ArrayList<>();
-                List<Schedule> schedules = processService.queryReleaseSchedulerListByProcessDefinitionCode(command.getProcessDefinitionCode());
-                listDate.addAll(CronUtils.getSelfFireDateList(start, end, schedules));
-                int listDateSize = listDate.size();
-                createCount = listDate.size();
-                if (!CollectionUtils.isEmpty(listDate)) {
-                    if (expectedParallelismNumber != null && expectedParallelismNumber != 0) {
-                        createCount = Math.min(listDate.size(), expectedParallelismNumber);
-                        if (listDateSize < createCount) {
-                            createCount = listDateSize;
+                if(startDate != null && endDate != null){
```

Comment:
It's better to separate these out into a method like: createComplementCommandByCron and createComplementCommandByDateList, just a code improvement suggestion.